### PR TITLE
Terraform Init (#51)

### PR DESCRIPTION
**Pull request by SarahJaine**
**Description of change**
Create persistent infrastructure with Terraform.

**Reasoning behind RDS plan choices:** 
- `micro-psql` for dev and staging: smallest dedicated instance that allows us to use postgres v12.4
- `small-psql` for prod: creates a small-t3 instance, 1vCPU and 2 GiB RAM should be sufficient for now. We'll be better able to tune this to the appropriate size when we have prod deployed and a monitoring system set up. Performing an update to the instance size will cause only a brief period of downtime (second to minutes). A developer can easily and autonomously scale an instance to a larger plan, however, scaling an instance to a smaller plan requires us to email cloud.gov support.

**Note to reviewers:** This is infrastructure configuration, not code.  It is highly repetitive and not modular by design. The terraform block must be repeated for every environment and parameters for those tags do not accept variables as input.

**How to test**
- @jasalisbury helped me out with testing- we worked through terraform/README.md to create a rds instance for dev. You can [check out the RDS instance in cloud.gov](https://dashboard.fr.cloud.gov/services).

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/23

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated - **added new section to main README, created terraform/README.md**
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

**Comment by SarahJaine** [comment]
Successfully applied in dev
```
➜  dev git:(main) ✗ tf apply
data.cloudfoundry_space.space: Refreshing state... [id=26178bf7-6275-45c5-9972-35ed819cf7c9]
data.cloudfoundry_service.rds: Refreshing state... [id=dcfb1d43-f22c-42d3-962c-7ae04eda24e7]
cloudfoundry_service_instance.database: Refreshing state... [id=8228924a-efb3-465e-9e2e-60b57beaa3ec]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # cloudfoundry_service_instance.database will be created
  + resource "cloudfoundry_service_instance" "database" {
      + id               = (known after apply)
      + name             = "ttahub-dev"
      + recursive_delete = false
      + service_plan     = "5ed9f319-4c75-4851-9885-598a96b7febc"
      + space            = "26178bf7-6275-45c5-9972-35ed819cf7c9"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

cloudfoundry_service_instance.database: Creating...
...
cloudfoundry_service_instance.database: Creation complete after 6m42s [id=fc548fed-aaf2-4681-9c76-2d1d93a08892]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

File: README.md [hunk 1]
```diff
@@ -57,20 +53,24 @@ Yarn Commands
...
 
-Integration
------------
+## Infrastructure
+
+### Persistent vs Ephemeral Infrastructure
+
+The infrastructure used to run this application can be categorized into two distinct types: ephemeral and persistent.
+- **Ephemeral infrastructure** is all the infrastructure that is recreated each time the application is deployed. Ephemeral infrastructure includes the "application" (as defined in Cloud.gov), the EC2 instances that application runs on, and the routes that application utilizes. This infrastructure is defined and deployed to Cloud.gov by the CircleCI configuration.
+- **Persistent infrastructure** is all the infrastructure that remains constant and unchanged despite application deployments. Persistent infrastructure includes the database used in each development environment. This infrastructure is defined and instantiated on Cloud.gov by the Terraform configuration files. For more about Terraform see [terraform/README.md](.circleci/config.yml).
```
Comment (rahearn) [inline README.md:62]: This link goes to the wrong spot (but I love that it's a link and not just text)

```suggestion
- **Persistent infrastructure** is all the infrastructure that remains constant and unchanged despite application deployments. Persistent infrastructure includes the database used in each development environment. This infrastructure is defined and instantiated on Cloud.gov by the Terraform configuration files. For more about Terraform see [terraform/README.md](terraform/README.md).
```

File: terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
+# Terraform
+
+The persistent AWS infrastructure is deployed as Cloud.gov assets via Terraform. For a description of persistent infrastructure and how it differs from ephemeral infrastructure see the root README.md.
+
+These docs are verbose because this is technology with which developers will rarely interact. I suggest you settle in for a nice long read with your favorite drink of choice.
```
Comment (rahearn) [inline terraform/README.md:5]: 😂

File: terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      - If state is clean, `plan` will not display any changes.
+      - If state is not clean and you are working on a feature branch try merging in the main branch; your local terraform files might be behind the current/applied state. If `plan` still shows an unclean state, reach out to your fellow developers for clarification and advice on how to proceed. It is very likely that changes were accidentally applied and a developer is currently working on a fix.
+
+    ```bash
+    terraform plan
+    ```
+
+1. **Make changes and get feedback**
+
+    Make any needed changes to your local Terraform configuration files. Open a PR for those changes. In your PR include the output from `terraform plan` so reviewers can see what resources will be updated, created and destroyed.
```
Comment (rahearn) [inline terraform/README.md:135]: Love this idea.

**Review by rahearn — CHANGES_REQUESTED** [review]
Fantastic work. Just the one little link update and this is ready to go.

File: README.md (hunk 1 above)
Comment (SarahJaine) [reply README.md:62]: Ha, cp error there-- thanks for catching it!

**Review by rahearn — APPROVED** [review]

**Pull request merged**